app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,46 @@
 class PostsController < ApplicationController
   def posts
     @user = User.find(params[:id])
+    @post = Post.find(params[:id])
   end
 
   def post
     @post = Post.find(params[:id])
   end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def addcomment
+    @user = current_user
+    @post = Post.find(params[:id])
+    @comment = current_user.comments.new(text: params[:post][:comment])
+    puts "--------------------------------------#{params[:comment]}"
+    @comment.post_id = @post.id
+    @comment.save
+    redirect_to "/user/#{current_user.id}/post/#{@post.id}"
+  end
+
+  def addlike
+    @user = current_user
+    @post = Post.find(params[:id])
+    @like = @post.likes.new
+    @like.post_id = @post.id
+    @like.author_id = @user.id
+    @like.save
+    redirect_to "/user/#{current_user.id}/post/#{@post.id}"
+  end
```

Comment:
- [ ] There should be a `comments` and `likes` controller with the `create` method where you can create the comments and likes just like the `posts` controller here. The rails are designed in such a way that they will talk easily with the controller, views, and models.